src/terminator/intrinsic.rs

```diff
@@ -402,6 +402,25 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                 }
             }
 
+            "write_bytes" => {
+                let u8 = self.tcx.types.u8;
+                let ty = substs.type_at(0);
+                let size = self.type_size(ty)?.expect("write_bytes() type must be sized");
+                let val_byte = self.value_to_primval(arg_vals[1], u8)?.to_u128()?;
+                let mut pattern: u128 = 0;
+                for ii in 0..size {
+                    pattern |= val_byte << (8 * ii);
+                }
+                let val_full = Value::ByVal(PrimVal::from_u128(pattern));
+                let mut ptr = arg_vals[0].read_ptr(&self.memory)?;
+
+                let count = self.value_to_primval(arg_vals[2], usize)?.to_u64()?;
+                for _ in 0..count {
+                    self.write_value_to_ptr(val_full, ptr, ty)?;
```

Comment:
> I would rather eliminate all the pattern/val_full code and replace it with a simple loop with size * count iterations, writing a single val_byte each iteration.

Actually, I think you could do a lower-level `let bytes = self.memory.get_bytes_mut(ptr, size * count, ty_align);` and then do a loop simply assigning the bytes directly. This would be much more efficient than repeatedly calling a single-byte write function on `Memory` which would do a lot of checks.